elasticapm/utils/stacks.py

```diff
@@ -147,24 +147,43 @@ def iter_traceback_frames(tb):
     while tb:
         # support for __traceback_hide__ which is used by a few libraries
         # to hide internal frames.
-        f_locals = getattr(tb.tb_frame, 'f_locals', {})
+        frame = tb.tb_frame
+        f_locals = getattr(frame, 'f_locals', {})
         if not _getitem_from_frame(f_locals, '__traceback_hide__'):
-            yield tb.tb_frame, getattr(tb, 'tb_lineno', None)
+            yield frame, getattr(tb, 'tb_lineno', None)
         tb = tb.tb_next
 
 
-def iter_stack_frames(frames=None, skip=0):
+def iter_stack_frames(frames=None, skip=0, skip_top_modules=()):
     """
     Given an optional list of frames (defaults to current stack),
     iterates over all frames that do not contain the ``__traceback_hide__``
     local variable.
+
+    Frames can be skipped by either providing a number, or a tuple
+    of module names. If the module of a frame matches one of the names
+    (using `.startswith`, that frame will be skipped. This matching will only
+    be done until the first frame doesn't match.
```

Comment:
Glad you asked :D I actually intend to filter more than just `elasticapm.`. In https://github.com/elastic/apm-agent-python/pull/164, I introduce a `functools.partial` call. In CPython, this doesn't create a frame (I suppose because it's implemented in C? Not sure). But in PyPy, this creates a frame, with the module `_functools`. So my intention is to rebase that PR after merging this one, and then adding `_functools` to that list. But we only want to kill that one frame at the top, and not any `functool.partial` calls that come later on.

Another argument for not filtering out _all_ frames that match is that this should only be used to cut away the top frames of the stack trace that are introduced by stack collection itself. If you call into any `elasticapm` code in other parts of the stack, that should IMO not be suppressed.

Reply:
that makes perfect sense. i am wiser now.

Reply:
there it is: https://github.com/elastic/apm-agent-python/blob/46e59e688bf5fe44464327bad5c2c10cd470edd6/elasticapm/base.py#L132-L137